arduino-ide-extension/src/browser/boards/boards-service-provider.ts

```diff
@@ -41,402 +141,327 @@ export class BoardsServiceProvider
-        // it is just a FQBN, so we need to find the `selected` board among the `AvailableBoards`
-        const selectedAvailableBoard = AvailableBoard.is(selectedBoard)
-          ? selectedBoard
-          : this._availableBoards.find(
-              (availableBoard) =>
-                Board.hardwareIdEquals(availableBoard, selectedBoard) ||
-                Board.sameAs(availableBoard, selectedBoard)
-            );
-        if (
-          selectedAvailableBoard &&
-          selectedAvailableBoard.selected &&
-          selectedAvailableBoard.state === AvailableBoard.State.recognized
-        ) {
-          return;
-        }
-        this.logger.info(
-          `Board package ${event.item.id} was uninstalled. Discarding the FQBN of the currently selected ${selectedBoard.name} board.`
-        );
-        const selectedBoardWithoutFqbn = {
-          name: selectedBoard.name,
-          // No FQBN
-        };
-        this.boardsConfig = {
-          ...this.boardsConfig,
-          selectedBoard: selectedBoardWithoutFqbn,
-        };
-      }
+  updatePort(selectedPort: PortIdentifier | undefined): boolean {
+    const selectedBoard = this._boardsConfig.selectedBoard;
+    const previousSelectedPort = this._boardsConfig.selectedPort;
+    if (selectedPort && selectedBoard) {
+      this._boardListHistory[Port.keyOf(selectedPort)] = selectedBoard;
     }
-  }
-
-  protected tryReconnect(): boolean {
-    if (this.latestValidBoardsConfig && !this.canUploadTo(this.boardsConfig)) {
-      // ** Reconnect to a board unplugged from, and plugged back into the same port
-      for (const board of this.availableBoards.filter(
-        ({ state }) => state !== AvailableBoard.State.incomplete
-      )) {
-        if (
-          Board.hardwareIdEquals(
-            this.latestValidBoardsConfig.selectedBoard,
-            board
-          )
-        ) {
-          const { name, fqbn } = this.latestValidBoardsConfig.selectedBoard;
-          this.boardsConfig = {
-            selectedBoard: {
-              name: board.name === Unknown || !board.name ? name : board.name,
-              fqbn: board.fqbn || fqbn,
-              port: board.port,
-            },
-            selectedPort: board.port,
-          };
-          return true;
-        }
-
-        if (
-          this.latestValidBoardsConfig.selectedBoard.fqbn === board.fqbn &&
-          this.latestValidBoardsConfig.selectedBoard.name === board.name &&
-          Port.sameAs(this.latestValidBoardsConfig.selectedPort, board.port)
-        ) {
-          this.boardsConfig = this.latestValidBoardsConfig;
-          return true;
-        }
-      }
-      // **
-
-      // ** Reconnect to a board whose port changed due to an upload
-      if (!this.boardConfigToAutoSelect) return false;
-
-      this.boardsConfig = this.boardConfigToAutoSelect;
-      this.boardConfigToAutoSelect = undefined;
-      return true;
-      // **
+    this._boardsConfig.selectedPort = selectedPort;
```

Comment:
isn't this assignment redundant if `portIdentifierEquals(previousSelectedPort, selectedPort) === true` would it be more appropriate to do it after the "NOOP if block"?

Reply:
Thanks. The logic is entirely new now. See https://github.com/arduino/arduino-ide/pull/2165#discussion_r1295753958